### PR TITLE
Added Iteration model + some exception

### PR DESCRIPTION
**Pull request by BjarkeBP**
Model:
- Iteration
- IterationValidator

Exceptions:
- IterationTitleEmptyException
- IterationTitleTooLongException
- IterationTitleTooShortException

Also add spaces to UnitTests.csproj, idk why

### CODE REVIEW

File: src/Tests/UnitTests/models/Iteration/IterationTests.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    [Fact]
+    public void Create_ShouldReturnNewIterationInstance()
+    {
+        // Act
+        var iteration = Iteration.Create();
+
+        // Assert
+        Assert.NotNull(iteration);
+        Assert.IsType<Iteration>(iteration);
+        Assert.NotEqual(Guid.Empty, iteration.Uid);
```
Comment (MalasataXD) [inline src/Tests/UnitTests/models/Iteration/IterationTests.cs:16]: Remove... These are not necessary

File: src/Tests/UnitTests/models/Iteration/IterationTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    [Theory]
+    [InlineData("", true)]  // Invalid title: empty string
+    [InlineData("   ", true)]  // Invalid title: whitespace only
+    [InlineData("ab", true)]  // Invalid title: less than 3 characters
+    [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", true)]  // Invalid title: exceeds max length
+    [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", false)]  // Valid title: max length
+    [InlineData("ValidTitle", false)]  // Valid title
+    public void UpdateTitle_ShouldValidateCorrectly(string title, bool expectedFailure)
```
Comment (MalasataXD) [inline src/Tests/UnitTests/models/Iteration/IterationTests.cs:26]: Split it up 👍

File: src/Tests/UnitTests/models/Iteration/IterationTests.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        Assert.IsType<Iteration>(iteration);
+        Assert.NotEqual(Guid.Empty, iteration.Uid);
+    }
+
+    [Theory]
+    [InlineData("", true)]  // Invalid title: empty string
+    [InlineData("   ", true)]  // Invalid title: whitespace only
+    [InlineData("ab", true)]  // Invalid title: less than 3 characters
+    [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", true)]  // Invalid title: exceeds max length
+    [InlineData("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", false)]  // Valid title: max length
```
Comment (MalasataXD) [inline src/Tests/UnitTests/models/Iteration/IterationTests.cs:24]: A little creativity?

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public IterationTitleEmptyException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected IterationTitleEmptyException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs:34]: Remove.

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.iteration.iterationTitle;
+
+/// <summary>
+/// Exception for when a Iteration is created without a title.
+/// </summary>
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs:8]: Remove.

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleEmptyException.cs:1]: Remove

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public IterationTitleTooLongException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected IterationTitleTooLongException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs:34]: Remove.

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.iteration.iterationTitle;
+
+/// <summary>
+/// Exception for when a Iteration is created with a title that are too long.
+/// </summary>
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs:8]: Remove

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooLongException.cs:1]: Remove

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public IterationTitleTooShortException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected IterationTitleTooShortException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs:34]: Remove

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs [hunk 11]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.iteration.iterationTitle;
+
+/// <summary>
+/// Exception for when a Iteration is created with a too short title
+/// </summary>
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs:8]: Remove

File: src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs [hunk 12]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Iteration/IterationTitle/IterationTitleTooShortException.cs:1]: Remove

**Review by MalasataXD — CHANGES_REQUESTED** [review]
"Minor" changes. A little of the same 😉

File: src/Tests/UnitTests/models/Iteration/IterationTests.cs (hunk 3 above)
Comment (BjarkeBP) [reply src/Tests/UnitTests/models/Iteration/IterationTests.cs:24]: **NO**

**Review by MalasataXD — APPROVED** [review]
Nice work! It is ready to merge 👍

**Pull request merged**